### PR TITLE
MNT Make employee validatable

### PR DESCRIPTION
**Pull request by GuySartorelli**
Fix broken validation that's already there
Add email field for testing field validation

Required for https://github.com/silverstripe/silverstripe-admin/pull/1543

## Issue
- https://github.com/silverstripe/silverstripe-admin/issues/1539

### CODE REVIEW

File: code/Employee.php [hunk 1]
```diff
@@ -142,7 +145,7 @@ public function validate()
     {
         $result = parent::validate();
         if (!$this->Name) {
-            $result->error('"Name" can\'t be blank');
+            $result->addFieldError('Name', '"Name" can\'t be blank');
```
Comment (ChloeSartorelli) [inline code/Employee.php:148]: This was just plum broken. There's no such method as `error()` on `ValidationResult`.

File: code/Employee.php
Comment (sabina-talipova) [inline code/Employee.php:1]: Looks good. Could we check, please, if it appropriate, if child class already has something like the method below or user wants to override parent method.
 
```php
public function getCMSCompositeValidator(): CompositeValidator
{
    $validator = CompositeValidator::create([MyNewValidator::create(['MyRequiredField'])]);
    return $validator;
}
```

**Review by sabina-talipova — CHANGES_REQUESTED** [review]

File: code/Employee.php
Comment (ChloeSartorelli) [reply code/Employee.php:1]: Can you please explain why? This module only exists for us to create text fixture classes for behat tests... nobody is using or subclassing this.

**Review by sabina-talipova — APPROVED** [review]

File: code/Employee.php
Comment (sabina-talipova) [reply code/Employee.php:1]: Probably, example wasn't quite proper. My concern was, will this solution work if user decided somehow to override parent method and add another validator. 
But I agree, it is not a question of `frameworktest` module.

**Pull request merged**